src/utils/hooks/requires-site-info.mjs

```diff
@@ -0,0 +1,29 @@
+import { error, warn } from '../command-helpers.mjs'
+/**
+ * A preAction hook that errors out if siteInfo is an empty object
+ * @param {*} command
+ */
+const requiresSiteInfo = async (command) => {
+  const { api, site } = command.netlify
+  const siteId = site.id
+  if (!siteId) {
+    warn('Did you run `netlify link` yet?')
+    error(`You don't appear to be in a folder that is linked to a site`)
```

Comment:
nit: I always found it counterintuitive to have a function call actually terminating the process. I think having a `return` would make it clearer that the function call actually ends the flow.

```suggestion
    return error(`You don't appear to be in a folder that is linked to a site`)
```